### ISSUE TITLE
exporting a named function triggers no-undef

### ISSUE THREAD

**Issue by jochenberger** [eslint/eslint#7269]
The following ES6 module triggers the `no-undef` rule:
```javascript
export default square = (x) => x * x
```
The error is:
`  1:16  error  'square' is not defined  no-undef`

**Comment by eslintbot** [comment]
Hi @jochenberger, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by kaicataldo** [comment]
The variable `square` isn't declared anywhere in your example, but it's hard to know what's going on without some more info.

**Comment by jochenberger** [comment]
Apparently that syntax is not supported, so the issue is invalid.
